Add edge-19.6.2 release notes (#2941)
linkerd/linkerd2: CHANGES.md

## edge-19.6.2

* CLI
  * Added the `--linkerd-cni-enabled` flag to the `install` subcommands so that

>     `NET_ADMIN` capability is omitted from the control plane's PSP

Author: Oh no problem thanks for the clarification on that!

* Controller
  * Fixed security context values so that injection does not fail on
    restricted PSPs (thanks @codeman9!)

>   * Fixed webhook failure policy in order to avoid timing issues in HA mode

Reviewer: The reason for that change is to avoid un-injected applications from being deployed, if the proxy injector fails as a result of unexpected errors (like network timeout etc.). Something like:

_Default the webhook failure policy to "fail", to account for unexpected errors during auto-proxy injection. This ensures un-injected applications aren't deployed._

  * Introduced control plane's PSP and RBAC resources into Helm templates;
    these policies are only in effect if the PSP admission controller is
    enabled
  * Fixed MWC namespace value so that when installing multiple control planes,

>     there is a unique configuration for each one

Reviewer: User action is required here, if they are using the `linkerd upgrade` command to upgrade the latest edge. After the upgrade, they will see two mutating webhook configurations:

```
k get mutatingwebhookconfigurations.admissionregistration.k8s.io
NAME                                    CREATED AT
linkerd-proxy-injector-webhook-config   2019-06-13T22:22:55Z
linkerd-linkerd-proxy-injector-webhook-config 2019-06-13T22:22:55Z

```
They will need to manually remove the `linkerd-proxy-injector-webhook-config` mwc. This isn't relevant if they are doing a fresh installation of the edge release.

Author: Is this included in docs? I think this information is better conveyed through docs rather than the release notes

Reviewer: sgtm. yeah, probably not needed for an edge release.

  * Removed `UPDATE` operation from proxy-injector webhook because pod
    mutations are disallowed during update operations
* Proxy

>   * The `l5d-override-dst` header is now honored for inbound service profiles

Reviewer 2: ```suggestion
  * The `l5d-override-dst` header is now honored for inbound service profile discovery
```
might be a little bit clearer IMO?

>   * Added the prefix of metric reports in log lines

Reviewer 3: not user facing

>   * Added the prefix of metric reports in log lines

Reviewer 2: This is fairly unlikely to be user-facing IMO

>   * Fixed metrics so that response errors are properly counted

Reviewer 2: TIOLI: what do you think of 
```suggestion
  * Fixed response errors not being properly counted in metrics
```

Reviewer 3: _Fixed a bug where errors were not counted in `response_total` metrics_ ?

Author: _Include errors in `response_total` metrics_?

I'd like to avoid "fixed a bug ..." bullets

>   * Changed requests to services with no endpoints to fail faster

Reviewer 3: This is phrased awkwardly.  Perhaps: _Changed the load balancer to require that kubernetes services are resolved via the control plane._

Reviewer 2: Oh, I like that. @kleimkuhler and I bikeshedded over this a bit offline & had a hard time coming up with a concise description, but yours is much better IMO.

## edge-19.6.2

* CLI
  * Added the `--linkerd-cni-enabled` flag to the `install` subcommands so that
    `NET_ADMIN` capability is omitted from the CNI-enabled control plane's PSP
* Controller

>   * Fixed security context values so that injection does not fail on

Reviewer: Can we change this to something like _Defined least privilege default security context values for the proxy container so that auto-injection does not fail on..._?

Author: Wording a bit differently but sure!

    restricted PSPs (thanks @codeman9!)
  * Default the webhook failure policy to `Fail` in order to account for
    unexpected errors during auto-inject; this ensures uninjected applications
    are not deployed
  * Introduced control plane's PSP and RBAC resources into Helm templates;
    these policies are only in effect if the PSP admission controller is
    enabled
  * Fixed MWC namespace value so that when installing multiple control planes,
    there is a unique configuration for each one
  * Removed `UPDATE` operation from proxy-injector webhook because pod
    mutations are disallowed during update operations
* Proxy

>   * The `l5d-override-dst` header is now honored for inbound service profile

Reviewer 2: TIOLI, but personally I think 
```suggestion
  * The `l5d-override-dst` header is now used for inbound service profile
```
might be better
